specifyweb/frontend/js_src/lib/components/TreeView/Row.tsx

```diff
@@ -74,18 +74,41 @@ export function TreeRow<SCHEMA extends AnyTree>({
   const isExpanded = Array.isArray(conformation);
   const isLoading = isExpanded && !Array.isArray(rows);
   const displayChildren = isExpanded && typeof rows?.[0] === 'object';
+  const orderByField = userPreferences.get(
+    'treeEditor',
+    'behavior',
+    'orderByField'
+  );
+
   React.useEffect(() => {
     if (!isLoading) return undefined;
 
-    void getRows(row.nodeId).then((rows) =>
-      destructorCalled ? undefined : setRows(rows)
-    );
+    void getRows(row.nodeId).then((fetchedRows: RA<Row>) => {
+      const sortedRows = Array.from(fetchedRows).sort((a, b) => {
```

Comment:
(optional) we have a sortFunction utility that can make this a bit less verbose:

```ts
.sort(sortFunction(
  orderByField === 'rankID' ? (row)=>row.rankId :
  orderByField === 'nodeNumber' ? (row)=>row.nodeNumber :
  ...
))
```

Reply:
[`294e32f` (#6046)](https://github.com/specify/specify7/pull/6046/commits/294e32fa47be2199f7df55b9458cbaadbd8de1f8)